state-chain/pallets/cf-environment/src/lib.rs

```diff
@@ -576,6 +619,73 @@ pub mod pallet {
 			// Witness the agg_key rotation manually in the vaults pallet for assethub
 			T::AssethubVaultKeyWitnessedHandler::on_first_key_activated(tx_id.block_number)
 		}
+
+		// TODO: PRO-2554 - User should be charged a transaction fee
+		#[pallet::call_index(10)]
+		#[pallet::weight({
+			let (dispatch_weight, dispatch_class) = weight_and_dispatch_class::<T>(calls);
+			let dispatch_weight = dispatch_weight.saturating_add(T::WeightInfo::submit_unsigned_batch_runtime_call(calls.len() as u32));
+			(dispatch_weight, dispatch_class)
+		})]
+		pub fn submit_unsigned_batch_runtime_call(
+			origin: OriginFor<T>,
+			calls: Vec<<T as Config>::RuntimeCall>,
+			transaction_metadata: TransactionMetadata,
+			user_signature_data: UserSignatureData,
+		) -> DispatchResult {
```

Comment:
A couple of things:

- We don't need a Vec of calls here. We have a `batch` runtime call so it can just be a single call (which might be `batch`. 
- It follows from the above then that `atomic` doesn't need to be in the transaction metadata. 
- It's not really an unsigned call - it's signed with a non-native wallet, so calling it unsigned and then telling users to submit a signed call via submit_unsigned is a bit confusing. I would rename this to something like `non_native_signed_call` or even `submit_eip712_call`.